pkg/store/cache/caching_bucket_factory.go

```diff
@@ -104,10 +105,17 @@ func NewCachingBucketFromYaml(yamlContent []byte, bucket objstore.Bucket, logger
 	case string(GroupcacheBucketCacheProvider):
 		const basePath = "/_galaxycache/"
 
-		c, err = cache.NewGroupcache(logger, reg, backendConfig, basePath, r, bucket,
-			isTSDBChunkFile, isMetaFile, isBlocksRootDir,
-			config.MetafileExistsTTL, config.MetafileDoesntExistTTL, config.MetafileContentTTL,
-			config.ChunkObjectAttrsTTL, config.ChunkSubrangeTTL, config.BlocksIterTTL)
+		cfg.SetCacheImplementation(c)
+		// Configure cache.
+		cfg.CacheAttributes("chunks", c, isTSDBChunkFile, config.ChunkObjectAttrsTTL)
+		cfg.CacheGetRange("chunks", c, isTSDBChunkFile, config.ChunkSubrangeSize, config.ChunkObjectAttrsTTL, config.ChunkSubrangeTTL, config.MaxChunksGetRangeRequests)
+		cfg.CacheExists("meta.jsons", c, isMetaFile, config.MetafileExistsTTL, config.MetafileDoesntExistTTL)
+		cfg.CacheGet("meta.jsons", c, isMetaFile, int(config.MetafileMaxSize), config.MetafileContentTTL, config.MetafileExistsTTL, config.MetafileDoesntExistTTL)
+
+		// Cache Iter requests for root.
+		cfg.CacheIter("blocks-iter", c, isBlocksRootDir, config.BlocksIterTTL, JSONIterCodec{})
```

Comment:
we need to move this outside, after L90, and remove the duplicate code from L129 to L135

Reply:
Since our goal was to remove the duplicate code anyways.